### PR TITLE
Add a demo of LiH VQE

### PR DESCRIPTION
**Pull request by Hugh-888**

### CODE REVIEW

File: docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+#
+# ### 2) 哈密顿量分解与期望值重构
+# 2. 把 LiH 的哈密顿量写成 Pauli 基矢的线性组合：$H=\sum_k w_k O_k$，其中 $w_k$ 是分解系数。则哈密顿量期望值满足：
+# $$ <H> = \sum_k w_k <O_k> $$
+#
+# 在这里，每个 Pauli 基矢对应一个 6 比特算符（例如 `IIIZZZ`）。
+# 接着把它切分成两个 3 比特算符 $O_a$ 与 $O_b$，使得 $O_k=O_a\otimes O_b$。
+# 然后分别用 $U\psi_3^n$ 与 $V\psi_3^n$ 计算对应子系统上的算符期望值。
+#
+# $$<O> = <O_a><O_b> = \sum_{nm} \lambda_n \lambda_m <\psi_3^n|U^+O_a U|\psi_3^n><\psi_3^n|V^+O_b V|\psi_3^n>  = \vec{\lambda}^T M \vec{\lambda}$$
```
Comment (sansiro77) [inline docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py]: 这里的整体期望值不应写成 `<O_a><O_b>` 的简单乘积；forged 态下应直接写成 `<O_a \otimes O_b>` 的双重求和形式。并且当前求和是 `n,m`，但矩阵元仍写成同一个 `n`，和后面的 `M_{ij}` 定义也不一致。建议把这一式改成与矩阵定义一致的表达。

File: docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+# M_{00} & M_{01} & M_{02}\\
+# M_{10} & M_{11} & M_{12} \\
+# M_{20} & M_{21} & M_{22}
+# \end{pmatrix}$， 其中$M_{ij}=<\psi_3^i|U^+O_a U|\psi_3^j><\psi_3^i|V^+O_b V|\psi_3^j>$。由于哈密顿量是厄米算符，可以得到 $M$ 的厄米性：$M^\dagger = M$（其中 $\dagger$ 表示转置并取共轭）。
+#
+# ### 3) 最小特征值与基态能量
+# 3. 把所有算符的期望值按分解系数求和，可得到
+# $M_{sum}=\sum_k w_k M_k$。
+# 于是哈密顿量的期望值可以写成二次型：$\langle H\rangle = \vec{\lambda}^T M_{sum} \vec{\lambda}$。
+# 在约束 $||\vec{\lambda}||=1$ 下，这个二次型的最小值等于矩阵 $M_{sum}$ 的最小特征值；对应的 $\vec{\lambda}$ 就是该特征值的特征向量。因此，最小特征值也对应于哈密顿量的基态能量。
```
Comment (sansiro77) [inline docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py]: 这里“最小特征值也对应于基态能量”的说法少了常数项说明。代码实际优化的是 `test[0] + constant`，因此若 `M_sum` 只收集 Pauli 非常数项，理论文字应明确总能量还需要再加哈密顿量分解中的 `constant`。

File: docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py [hunk 3]
```diff
@@ -0,0 +1,580 @@
...
+# ### 叠加态制备门序列
+# 对于叠加态 $\psi_{R}$ 和 $\psi_{I}$ 的制备：在光量子电路中，我们通常先制备一个 Fock 初态，然后再作用“叠加态变换矩阵”，即可得到相应的叠加态。
+# 在实现中，这个叠加态变换往往可以与后续 Ansatz 的酉变换合并，从而减少电路深度。
+# 下面给出 6 个叠加态所对应的构造顺序。涉及的基本门 $H$、$S$、$SWAP$、$CNOT$ 的矩阵形式如下。
+# $$
+# H = \frac{\sqrt{2}}{2}\begin{pmatrix}
+# 1 & 1\\
+# 1 & -1
+# \end{pmatrix},
+# S = \frac{\sqrt{2}}{2}\begin{pmatrix}
```
Comment (sansiro77) [inline docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py]: 这里给出的 `S` 矩阵和代码里的 `s_mat = diag(1, i)` 不一致。当前公式写成了带 `1/sqrt(2)` 的非标准矩阵，会误导后面的叠加态制备推导；建议改成标准相位门 `diag(1, i)`。

File: docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+        dic = pickle.load(f)
+    constant = dic['constant']
+    pauli_list_full = dic['pauli_list']
+    energy_full = dic['fci_energy']
+    u_measure_list_q3_ts, idx_z_list_q3_ts, pauli_coef = get_measure_u(pauli_list_full)
+
+    loss_all = []
+    pauli_coef_ts = torch.tensor(pauli_coef)
+    data = nn.Parameter(torch.rand(64) * 2 * torch.pi)
+    lambdas = nn.Parameter(torch.rand(1, 3, dtype=torch.complex128))
```
Comment (sansiro77) [inline docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.py]: `lambdas` 被加入 optimizer，但后续 loss 计算没有使用它；当前实现已经通过 `torch.linalg.eigh()` 直接取最小特征值，因此这个参数不会参与训练。建议删除它，或改成显式使用/展示 eigensolver 返回的 Schmidt 系数，避免和前文叙述冲突。

**Review by sansiro77 — COMMENTED** [review]
内容 review：主要是理论表述与代码行为一致性问题。

**Comment by sansiro77** [comment]
补充一个文档导航问题：新增了 `docs/source/demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.ipynb`，但 `docs/source/index.md` 的 Demos toctree 里还没有加入该路径。这样文档站点不会在导航中展示这个新 demo，建议补上 `demos/difficult/vqe_lih/tutorial_LiH_molecule_vqe.ipynb`。
